front/pages/w/[wId]/builder/data-sources/[name]/upsert.tsx

```diff
@@ -118,8 +118,19 @@ export default function DataSourceUpsert({
 
   const handleFileLoadedText = (e: any) => {
     const content = e.target.result;
-    setText(content);
     setUploading(false);
+
+    // Enforce plan limits: DataSource documents size.
+    if (
+      owner.plan.limits.dataSources.documents.sizeMb != -1 &&
+      text.length > 1024 * 1024 * owner.plan.limits.dataSources.documents.sizeMb
+    ) {
+      window.alert(
+        "DataSource document upload size is limited to 1MB (of raw text). Contact team@dust.tt if you want to increase this limit."
```

Comment:
NIT: instead of 1MB should we do:

```javascript
window.alert(
        `DataSource document upload size is limited to ${owner.plan.limits.dataSources.documents.sizeMb}MB (of raw text). Contact team@dust.tt if you want to increase this limit.`
``` 

(same below)

Reply:
Yep will change :+1: